File: frontend/civ_hex/lib/models/player.dart
class Player {
  String wsId;
  String currentGameId;
  String playerName;
  num victoryPoints;
  num population;
  num ownedTileCount;
  num wood;
  num stone;
  num grain;
  num sheep;
  num gold;


  Player({
    required this.wsId,
    required this.currentGameId,
    required this.playerName,
    required this.victoryPoints,
    required this.population,
    required this.ownedTileCount,
    required this.wood,
    required this.stone,
    required this.grain,
    required this.sheep,
    required this.gold

  });

  factory Player.fromJson(Map<String, dynamic> json) {
    return Player(
        wsId: json['WsId'],
        currentGameId: json['CurrentGameId'],
        playerName: json['PlayerName'],
        victoryPoints: json['VictoryPoints'],
        population: json['Population'],
        ownedTileCount: json['OwnedTileCount'],
        wood: json['Wood'],
        stone: json['Stone'],
        grain: json['Grain'],
        sheep: json['Sheep'],
        gold: json['Gold']
    );
  }
}

File: backend/Models/entities/Player.cs
namespace backend.Models.entities;

public class Player
{
    #region Properties

    public Guid WsId { get; set; }
    public Guid CurrentGameId { get; set; }
    public string PlayerName { get; set; }
    public int VictoryPoints { get; set; }
    public int Population { get; set; }
    public int OwnedTileCount { get; set; }
    public int Gold { get; set; }
    public int Wood { get; set; }
    public int Stone { get; set; }
    public int Grain { get; set; }
    public int Sheep { get; set; }

    #endregion
    
    



    public Player(string playerName, int gold, Guid wsId, Guid currentGameId)
    {
        PlayerName = playerName;
        Gold = gold;
        WsId = wsId;
        CurrentGameId = currentGameId;
    }


}